File: winter-compiler/src/test/resources/WithInjectedGenericFields_WinterMembersInjector.java
package test;

import io.jentz.winter.GenericClassTypeKey;
import io.jentz.winter.Graph;
import io.jentz.winter.inject.MembersInjector;
import java.lang.Integer;
import java.lang.Override;
import java.lang.String;
import java.util.List;
import java.util.Map;
import javax.annotation.Generated;

@Generated(
        value = "io.jentz.winter.compiler.WinterProcessor",
        date = "2019-02-10T14:52Z"
)
public final class WithInjectedGenericFields_WinterMembersInjector implements MembersInjector<WithInjectedGenericFields> {
    @Override
    public void inject(final Graph graph, final WithInjectedGenericFields target) {
        target.field0 = graph.instanceByKey(new GenericClassTypeKey<Map<String, Integer>>(null) {});
        target.field1 = graph.instanceOrNullByKey(new GenericClassTypeKey<List<Integer>>(null) {});
    }
}

File: winter-compiler/src/test/resources/WithInjectedGenericFields_WinterMembersInjector.kt
package test

import io.jentz.winter.Graph
import io.jentz.winter.inject.MembersInjector
import javax.annotation.Generated
import kotlin.Int
import kotlin.String
import kotlin.collections.List
import kotlin.collections.Map

@Generated(
  value = ["io.jentz.winter.compiler.WinterProcessor"],
  date = "2019-02-10T14:52Z"
)
class WithInjectedGenericFields_WinterMembersInjector : MembersInjector<WithInjectedGenericFields> {
  override fun inject(graph: Graph, target: WithInjectedGenericFields) {
    target.field0 = graph.instance<Map<String, Int>>(generics = true)
    target.field1 = graph.instanceOrNull<List<Int>>(generics = true)
  }
}
